orc8r/cloud/go/services/configurator/storage/sql_entity_load_helpers.go

```diff
@@ -211,44 +220,45 @@ func (store *sqlConfiguratorStorage) getBuilder(networkID string, filter EntityL
 
 	where := sq.And{isInNetwork}
 	// Type, key filters
-	if filter.KeyFilter != nil {
-		where = append(where, sq.Eq{entCol(entKeyCol): filter.KeyFilter.Value})
+	if filterCopy.KeyFilter != nil {
+		where = append(where, sq.Eq{entCol(entKeyCol): filterCopy.KeyFilter.Value})
 	}
-	if filter.TypeFilter != nil {
-		where = append(where, sq.Eq{entCol(entTypeCol): filter.TypeFilter.Value})
+	if filterCopy.TypeFilter != nil {
+		where = append(where, sq.Eq{entCol(entTypeCol): filterCopy.TypeFilter.Value})
 	}
 	// Specific page
-	if criteria.PageToken != "" {
+	if criteriaCopy.PageToken != "" {
 		where = append(where, sq.Gt{entCol(entKeyCol): pageToken.LastIncludedEntity})
 	}
 
 	b := builder.Where(where)
 	return addSuffix(b), nil
 }
 
-func scanEntityRow(rows *sql.Rows, criteria EntityLoadCriteria) (NetworkEntity, error) {
+func scanEntityRow(rows *sql.Rows, criteria *EntityLoadCriteria) (*NetworkEntity, error) {
```

Comment:
Maybe there's some master reason why protobuf maintainers did it this way, but man this seems suspect lol. Like, we can't even pass a proto by value anymore?

Reply:
Yes, somewhat mean... It fails the `go vet -copylocks` check, since in API v2 implementation proto messages contain a `sync.Mutex`.

This comment in a related issue https://github.com/golang/protobuf/issues/1225#issuecomment-713699536 might clarify to some degree.